src/cephadm/cephadm

```diff
@@ -3597,22 +3615,46 @@ class Zypper(Packager):
         self.install(['podman'])
 
 
-def create_packager(stable=None, branch=None, commit=None):
-    distro, version, codename = get_distro()
+def create_packager(stable=None, version=None, branch=None, commit=None):
+    distro, distro_version, distro_codename = get_distro()
     if distro in YumDnf.DISTRO_NAMES:
-        return YumDnf(stable=stable, branch=branch, commit=commit,
-                   distro=distro, version=version)
+        return YumDnf(stable=stable, version=version,
+                      branch=branch, commit=commit,
+                   distro=distro, distro_version=distro_version)
     elif distro in Apt.DISTRO_NAMES:
-        return Apt(stable=stable, branch=branch, commit=commit,
-                   distro=distro, version=version, codename=codename)
+        return Apt(stable=stable, version=version,
+                   branch=branch, commit=commit,
+                   distro=distro, distro_version=distro_version,
+                   distro_codename=distro_codename)
     elif distro in Zypper.DISTRO_NAMES:
-        return Zypper(stable=stable, branch=branch, commit=commit,
-                      distro=distro, version=version)
-    raise Error('Distro %s version %s not supported' % (distro, version))
+        return Zypper(stable=stable, version=version,
+                      branch=branch, commit=commit,
+                      distro=distro, distro_version=distro_version)
+    raise Error('Distro %s version %s not supported' % (distro, distro_version))
 
 
 def command_add_repo():
-    pkg = create_packager(stable=args.release, branch=args.dev,
+    if args.version:
+        try:
+            (x, y, z) = args.version.split('.')
+        except Exception as e:
+            raise Error('version must be in the form x.y.z (e.g., 15.2.0)')
+        relnames = {
+            '16': 'pacific',
+            '15': 'octopus',
+            '14': 'nautilus',
+            '13': 'mimic',
+            '12': 'luminous',
+        }
+        if not args.release:
+            args.release = relnames.get(x, None)
+        if not args.release:
+            raise Error('unknown release %s (not in %s)' % (
+                x, ' '.join(relnames.values())))
```

Comment:
verify `release` matches `version` major?  for example: `--release octopus` and  `--version 14.2.1`

.... or just let this fail during shaman query because of an unknown uri?